chain-impl-mockchain/src/rewards.rs

```diff
@@ -163,6 +224,25 @@ pub fn rewards_contribution_calculation(epoch: Epoch, params: &Parameters) -> Va
     }
 }
 
+// Added new function to ensure rewards calculation complies with research results and commercial needs -- KH
```

Comment:
this comment doesn't need to be there. chain-libs is agnostic to commercial needs, and will make sure to support every uses.

Reply:
OK.  It was just an explanation for why this section of code had been added.